common/src/main/java/io/novafoundation/nova/common/validation/InputValidationMixinFactory.kt

```diff
@@ -0,0 +1,25 @@
+package io.novafoundation.nova.common.validation
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class InputValidationMixinFactory<P, E>(
+    private val validationSystem: ValidationSystem<P, E>,
+) {
+    fun create(inputFlow: Flow<String>, payloadFormatter: (String) -> P): InputValidationMixin<P, E> {
+        return InputValidationMixin(validationSystem, inputFlow, payloadFormatter)
+    }
+}
```

Comment:
To be honest I think using a validation system is overkill for field validations and it will lead to just more boilerplate.
We can come up with a simpler version that is easier to configure:
``` kotlin
interface FieldValidator {
   validate(inputStream: Flow<String>): Flow<FieldValidationResult>

  compaction object;
}

fun FieldValidator.oneOf(vararg inner: FieldValidator): FieldValidator

sealed class FieldValidationResult {
   object Ok: FieldValidationResult()

  class Error(val reason: String): FieldValidationResult()

class Warning(val reason: String): FieldValidationResult()
}
```